test/lib/mocked-dockerode.ts

```diff
@@ -1,4 +1,49 @@
+process.env.DOCKER_HOST = 'unix:///your/dockerode/mocks/are/not/working';
+
 import * as dockerode from 'dockerode';
+import { Stream } from 'stream';
+import _ = require('lodash');
+
+const overrides: Dictionary<(...args: any[]) => Resolvable<any>> = {};
+
+type DockerodeFunction = keyof dockerode;
+for (const fn of Object.getOwnPropertyNames(dockerode.prototype)) {
```

Comment:
Do you specifically want non-enumerable properties as well as enumerable?

Reply:
How do you mean? I want every function on the prototype...